internal/release/service.go

```diff
@@ -162,27 +162,21 @@ func (s *service) processFilters(ctx context.Context, filters []*domain.Filter,
 
 		l.Info().Msgf("Matched '%s' (%s) for %s", release.TorrentName, release.FilterName, release.Indexer)
 
-		// save release here to only save those with rejections from actions instead of all releases
-		if release.ID == 0 {
-			release.FilterStatus = domain.ReleaseStatusFilterApproved
-
-			if err = s.Store(ctx, release); err != nil {
-				l.Error().Err(err).Msgf("release.Process: error writing release to database: %+v", release)
-				return err
-			}
-		}
-
 		// found matching filter, lets find the filter actions and attach
 		actions, err := s.actionSvc.FindByFilterID(ctx, f.ID)
 		if err != nil {
 			s.log.Error().Err(err).Msgf("release.Process: error finding actions for filter: %s", f.Name)
 			return err
 		}
 
-		// if no actions, continue to next filter
-		if len(actions) == 0 {
-			s.log.Warn().Msgf("release.Process: no actions found for filter '%s', trying next one..", f.Name)
-			return nil
+		// if no actions, or all actions are disabled, continue to next filter
+		if len(actions) == 0 || !s.hasEnabledAction(actions) {
+			msg := "no actions found"
+			if len(actions) > 0 {
```

Comment:
I think it would be a better option to add a new optional parameter to the repo method `FindByFilterID(ctx context.Context, filterID int, active bool) ([]*Action, error)` which is called by `actionSvc.FindByFilterID`.

Then keep this as before with the `len(actions) == 0` check and change the message to `release.Process: no active actions found for filter '%s', trying next one..`.

Reply:
Done 👍